### PR TITLE
[Cleanup]: remove upgrade_state_dict logic, normformer, unused alignment/encoder args

### PR DESCRIPTION
**Pull request by suchenzang**
Tackling https://github.com/facebookresearch/metaseq/issues/165

Only checkpoints to work with here are with OPT models so far.

More cleanup around unused args / logic in preparation for reducing complexity around configuration / checkpointing logic.

* Removed all normformer args & related classes (sync ln)
* Removed stable embeddings 
* Removed FeedForwardNetwork class

### CODE REVIEW

**Comment by suchenzang** [comment]
More testing to come. Noting here:

```
shard0 = torch.load('/data/***/175B/raw/checkpoint_last-model_part-0-shard0.pt', map_location=torch.device("cpu"))

>>> shard0['shard_metadata']['param_metadata'][0]
{'fsdp_path': '', 'params': {'flat_param_0': {'names': ['decoder.embed_tokens.weight', 'decoder.embed_positions.weight', 'decoder.layer_norm.weight', 'decoder.layer_norm.bias'], 'shapes': [(6284, 12288), (2050, 12288), (12288,), (12288,)], 'numels': [77217792, 25190400, 12288, 12288], 'padding': 0}}, 'no_broadcast_optim_state': False, 'shared_param_info': [('decoder.embed_tokens.weight', 'decoder.output_projection.weight')]}

>>> shard0['shard_metadata']['param_metadata'][1]
{'fsdp_path': 'decoder.layers.0', 'params': {'flat_param_0': {'names': ['self_attn.qkv_proj.weight', 'self_attn.qkv_proj.bias', 'self_attn.out_proj.weight', 'self_attn.out_proj.bias', 'self_attn_layer_norm.weight', 'self_attn_layer_norm.bias', 'fc1.weight', 'fc1.bias', 'fc2.weight', 'fc2.bias', 'final_layer_norm.weight', 'final_layer_norm.bias'], 'shapes': [(4608, 12288), (4608,), (12288, 1536), (12288,), (12288,), (12288,), (6144, 12288), (6144,), (12288, 6144), (12288,), (12288,), (12288,)], 'numels': [56623104, 4608, 18874368, 12288, 12288, 12288, 75497472, 6144, 75497472, 12288, 12288, 12288], 'padding': 0}}, 'no_broadcast_optim_state': False, 'shared_param_info': []}

>>> shard0['shard_metadata']['param_metadata'][2]
{'fsdp_path': 'decoder.layers.1', 'params': {'flat_param_0': {'names': ['self_attn.qkv_proj.weight', 'self_attn.qkv_proj.bias', 'self_attn.out_proj.weight', 'self_attn.out_proj.bias', 'self_attn_layer_norm.weight', 'self_attn_layer_norm.bias', 'fc1.weight', 'fc1.bias', 'fc2.weight', 'fc2.bias', 'final_layer_norm.weight', 'final_layer_norm.bias'], 'shapes': [(4608, 12288), (4608,), (12288, 1536), (12288,), (12288,), (12288,), (6144, 12288), (6144,), (12288, 6144), (12288,), (12288,), (12288,)], 'numels': [56623104, 4608, 18874368, 12288, 12288, 12288, 75497472, 6144, 75497472, 12288, 12288, 12288], 'padding': 0}}, 'no_broadcast_optim_state': False, 'shared_param_info': []}

>>> len(shard0['shard_metadata']['param_metadata'])
97

>>> shard0['shard_metadata']['param_metadata'][-1]
{'fsdp_path': 'decoder.layers.95', 'params': {'flat_param_0': {'names': ['self_attn.qkv_proj.weight', 'self_attn.qkv_proj.bias', 'self_attn.out_proj.weight', 'self_attn.out_proj.bias', 'self_attn_layer_norm.weight', 'self_attn_layer_norm.bias', 'fc1.weight', 'fc1.bias', 'fc2.weight', 'fc2.bias', 'final_layer_norm.weight', 'final_layer_norm.bias'], 'shapes': [(4608, 12288), (4608,), (12288, 1536), (12288,), (12288,), (12288,), (6144, 12288), (6144,), (12288, 6144), (12288,), (12288,), (12288,)], 'numels': [56623104, 4608, 18874368, 12288, 12288, 12288, 75497472, 6144, 75497472, 12288, 12288, 12288], 'padding': 0}}, 'no_broadcast_optim_state': False, 'shared_param_info': []}

>>> shard0['shard_metadata']['param_metadata'][-2]
{'fsdp_path': 'decoder.layers.94', 'params': {'flat_param_0': {'names': ['self_attn.qkv_proj.weight', 'self_attn.qkv_proj.bias', 'self_attn.out_proj.weight', 'self_attn.out_proj.bias', 'self_attn_layer_norm.weight', 'self_attn_layer_norm.bias', 'fc1.weight', 'fc1.bias', 'fc2.weight', 'fc2.bias', 'final_layer_norm.weight', 'final_layer_norm.bias'], 'shapes': [(4608, 12288), (4608,), (12288, 1536), (12288,), (12288,), (12288,), (6144, 12288), (6144,), (12288, 6144), (12288,), (12288,), (12288,)], 'numels': [56623104, 4608, 18874368, 12288, 12288, 12288, 75497472, 6144, 75497472, 12288, 12288, 12288], 'padding': 0}}, 'no_broadcast_optim_state': False, 'shared_param_info': []}
```
```
>>> shard0['model'].keys()
dict_keys(['flat_param_0', 'decoder.version', 'decoder.layers.0.flat_param_0', 'decoder.layers.1.flat_param_0', 'decoder.layers.2.flat_param_0', 'decoder.layers.3
.flat_param_0', 'decoder.layers.4.flat_param_0', 'decoder.layers.5.flat_param_0', 'decoder.layers.6.flat_param_0', 'decoder.layers.7.flat_param_0', 'decoder.layer
s.8.flat_param_0', 'decoder.layers.9.flat_param_0', 'decoder.layers.10.flat_param_0', 'decoder.layers.11.flat_param_0', 'decoder.layers.12.flat_param_0', 'decoder
.layers.13.flat_param_0', 'decoder.layers.14.flat_param_0', 'decoder.layers.15.flat_param_0', 'decoder.layers.16.flat_param_0', 'decoder.layers.17.flat_param_0',
'decoder.layers.18.flat_param_0', 'decoder.layers.19.flat_param_0', 'decoder.layers.20.flat_param_0', 'decoder.layers.21.flat_param_0', 'decoder.layers.22.flat_pa
ram_0', 'decoder.layers.23.flat_param_0', 'decoder.layers.24.flat_param_0', 'decoder.layers.25.flat_param_0', 'decoder.layers.26.flat_param_0', 'decoder.layers.27
.flat_param_0', 'decoder.layers.28.flat_param_0', 'decoder.layers.29.flat_param_0', 'decoder.layers.30.flat_param_0', 'decoder.layers.31.flat_param_0', 'decoder.l
ayers.32.flat_param_0', 'decoder.layers.33.flat_param_0', 'decoder.layers.34.flat_param_0', 'decoder.layers.35.flat_param_0', 'decoder.layers.36.flat_param_0', 'd
ecoder.layers.37.flat_param_0', 'decoder.layers.38.flat_param_0', 'decoder.layers.39.flat_param_0', 'decoder.layers.40.flat_param_0', 'decoder.layers.41.flat_para
m_0', 'decoder.layers.42.flat_param_0', 'decoder.layers.43.flat_param_0', 'decoder.layers.44.flat_param_0', 'decoder.layers.45.flat_param_0', 'decoder.layers.46.f
lat_param_0', 'decoder.layers.47.flat_param_0', 'decoder.layers.48.flat_param_0', 'decoder.layers.49.flat_param_0', 'decoder.layers.50.flat_param_0', 'decoder.lay
ers.51.flat_param_0', 'decoder.layers.52.flat_param_0', 'decoder.layers.53.flat_param_0', 'decoder.layers.54.flat_param_0', 'decoder.layers.55.flat_param_0', 'dec
oder.layers.56.flat_param_0', 'decoder.layers.57.flat_param_0', 'decoder.layers.58.flat_param_0', 'decoder.layers.59.flat_param_0', 'decoder.layers.60.flat_param_
0', 'decoder.layers.61.flat_param_0', 'decoder.layers.62.flat_param_0', 'decoder.layers.63.flat_param_0', 'decoder.layers.64.flat_param_0', 'decoder.layers.65.fla
t_param_0', 'decoder.layers.66.flat_param_0', 'decoder.layers.67.flat_param_0', 'decoder.layers.68.flat_param_0', 'decoder.layers.69.flat_param_0', 'decoder.layer
s.70.flat_param_0', 'decoder.layers.71.flat_param_0', 'decoder.layers.72.flat_param_0', 'decoder.layers.73.flat_param_0', 'decoder.layers.74.flat_param_0', 'decod
er.layers.75.flat_param_0', 'decoder.layers.76.flat_param_0', 'decoder.layers.77.flat_param_0', 'decoder.layers.78.flat_param_0', 'decoder.layers.79.flat_param_0'
, 'decoder.layers.80.flat_param_0', 'decoder.layers.81.flat_param_0', 'decoder.layers.82.flat_param_0', 'decoder.layers.83.flat_param_0', 'decoder.layers.84.flat_
param_0', 'decoder.layers.85.flat_param_0', 'decoder.layers.86.flat_param_0', 'decoder.layers.87.flat_param_0', 'decoder.layers.88.flat_param_0', 'decoder.layers.
89.flat_param_0', 'decoder.layers.90.flat_param_0', 'decoder.layers.91.flat_param_0', 'decoder.layers.92.flat_param_0', 'decoder.layers.93.flat_param_0', 'decoder
.layers.94.flat_param_0', 'decoder.layers.95.flat_param_0'])
```

**Comment by suchenzang** [comment]
Breaking this PR down into separate ones.
* https://github.com/facebookresearch/metaseq/pull/229 for [6b424be](https://github.com/facebookresearch/metaseq/pull/197/commits/6b424be6e4b0dd734ac88d27dc3f3332a8c9a2bf)^..[8c01f2](https://github.com/facebookresearch/metaseq/pull/197/commits/28c01f2c6e992462908172d9fbf4e5e87ce14892)

File: metaseq/models/base_model.py [hunk 1]
```diff
@@ -214,54 +168,6 @@ def apply_prepare_for_onnx_export_(module):
...
-            moe_disable_padding=moe_disable_padding,
-            skip_prepare_for_inference=skip_prepare_for_inference,
-        )
-
-    @classmethod
-    def hub_models(cls):
-        return {}
-
 
 class EncoderDecoderModel(BaseModel):
```
Comment (stephenroller) [inline metaseq/models/base_model.py:172]: Why not axe EncDec entirely?

File: metaseq/models/transformer.py [hunk 2]
```diff
@@ -363,17 +251,13 @@ def __init__(self, args, dictionary, embed_tokens, no_encoder_attn=False):
             self.dropout_module = None
 
         self.share_input_output_embed = args.share_decoder_input_output_embed
-
```
Comment (stephenroller) [inline metaseq/models/transformer.py:366]: I kind of like this white space

File: metaseq/models/transformer_lm.py [hunk 3]
```diff
@@ -96,28 +96,6 @@ class TransformerLanguageModelConfig(MetaseqDataclass):
             )
         },
     )
-    use_stable_embedding: Optional[bool] = field(
```
Comment (stephenroller) [inline metaseq/models/transformer_lm.py:99]: We may yet regret this

File: metaseq/modules/transformer_layer.py [hunk 4]
```diff
@@ -37,65 +37,26 @@ def _ffn(x, fc1, activation_fn, fc2, dropout_module, ffn_ln=None):
         assert fc1.skip_bias_add
         x, bias_fc1 = fc1(x)
         x = fused_bias_gelu(x, bias_fc1)
-        if ffn_ln is not None:
```
Comment (stephenroller) [inline metaseq/modules/transformer_layer.py:40]: We may also regret this. Sync offline

File: metaseq/modules/transformer_layer.py [hunk 5]
```diff
@@ -37,65 +37,26 @@ def _ffn(x, fc1, activation_fn, fc2, dropout_module, ffn_ln=None):
...
         x = activation_fn(x)
-        if ffn_ln is not None:
-            x = ffn_ln(x)
         x = fc2(x)
     x = x.view(x_shape)
     x = dropout_module(x)
     return x
 
 
-class FeedForwardNetwork(nn.Module):
```
Comment (stephenroller) [inline metaseq/modules/transformer_layer.py:68]: This seems weird to delete

**Review by stephenroller — COMMENTED** [review]
Let's discuss with @ngoyal2707. We don't want to interrupt the two workstreams he and I have going.

File: metaseq/modules/transformer_layer.py (hunk 5 above)
Comment (suchenzang) [reply metaseq/modules/transformer_layer.py:68]: unused

File: metaseq/models/base_model.py (hunk 1 above)
Comment (suchenzang) [reply metaseq/models/base_model.py:172]: Last I tried there were some extra dependencies (mainly in tests for... sequence scoring? something else that we were keeping around)

File: metaseq/models/transformer_lm.py (hunk 3 above)
Comment (suchenzang) [reply metaseq/models/transformer_lm.py:99]: Do you want to turn it on in the next run?

File: metaseq/modules/transformer_layer.py (hunk 4 above)
Comment (suchenzang) [reply metaseq/modules/transformer_layer.py:40]: This was brought in with normformer which we turned off entirely for OPT. Do you want to add this one LN back?

File: metaseq/models/base_model.py (hunk 1 above)
Comment (suchenzang) [reply metaseq/models/base_model.py:172]: Ah, it's just tests (mentioned here: https://github.com/facebookresearch/metaseq/blob/main/docs/history.md) - I'll rip these out

**Pull request closed without merging**